orodha-keycloak/orodha_keycloak/connection.py

```diff
@@ -2,14 +2,84 @@
 This Module contains the OrodhaKeycloakClient class which is a facade
 used to interact with a keycloak server via python-keycloak.
 """
+import os
 import orodha_keycloak.connections.admin
 import orodha_keycloak.connections.client
-from orodha_keycloak.connections.exceptions import InvalidConnectionException
+from orodha_keycloak.exceptions import InvalidConnectionException
+
+
+class OrodhaCredentials:
+    """
+    A class used to obtain and package credentials for the Orodha system.
+    If kwargs are not passed in, The credentials are pulled from the environment.
+
+    Args(as kwargs):
+        orodha_keycloak_client_server(str): The url of the server that our keycloak is hosted at
+        orodha_keycloak_realm_name(str): The name of the keycloak realm that we are attempting to access.
+        orodha_keycloak_client_id(str): The keycloak client_id that we are using for the connection.
+        orodha_keycloak_client_secret_key(str): The secret key of the keycloak client.
+        orodha_keycloak_username(str) - Optional: The username of the user being impersonated by python-keycloak
+        orodha_keycloak_password(str) - Optional: The password of the user being impersonated by python-keycloak
+
+    Raises:
+        InvalidConnectionException: If required arguments are not available
+            as kwargs or in the environment.
+    """
+
+    def __init__(self, **kwargs):
+        arg_dict = {
+            "orodha_keycloak_client_server": kwargs.get("orodha_keycloak_client_server"),
```

Comment:
@Edison-Stuart It's kind of verbose to have `orodha_keycloak` on the keyword args. Try this:
```arg_dict = {
            "orodha_keycloak_client_server": kwargs.get("client_server"),
            "orodha_keycloak_realm_name": kwargs.get("realm_name"),
            "orodha_keycloak_client_id": kwargs.get("client_id"),
            "orodha_keycloak_client_secret_key": kwargs.get("client_secret_key"),
            "orodha_keycloak_username": kwargs.get("username"),
            "orodha_keycloak_password": kwargs.get("password")
        }
```